### PR TITLE
[SMP-2308] [SMP-2363] SMP Self-Managed Amazon ElastiCache Redis database tutorial

### PR DESCRIPTION
**Pull request by brian-f-harness**
SMP Self-Managed Amazon ElastiCache Redis database tutorial

For reviewers, a preview is available.

[External self-managed AWS ElastiCache](https://659d69fcd0b9190f3d787b7c--harness-developer.netlify.app/tutorials/self-managed-enterprise-edition/use-aws-elasticache)

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+You must have permissions to create an an ElastiCache instance, security groups, and subnet groups.
+
+## Set up Amazon ElastiCache for Redis
+
+To set up Amazon ElastiCache for Redis, do the following:
+
+1. Complete the steps in the [Getting started with Amazon ElastiCache for Redis](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/GettingStarted.html) guide in the AWS documentation. 
+
+:::info important
+Make you disable the cluster mode.
```
Comment (kapilgarg1996) [inline tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:55]:
```suggestion
Make sure you disable the cluster mode.
```

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Self-Managed Enterprise Edition requires you to install a database by default. You can optionally use an external database with your Self-Managed Enterprise Edition installation. This enables you to separate your data from node execution. To use an Amazon ElastiCache Redis database with your Harness Self-Managed Enterprise Edition installation, you must ensure that your hardware, software, and network meet the minimum requirements for installation and configuration.
+
+This tutorial describes how to configure an Amazon ElastiCache Redis database for your installation.
+
+## Hardware requirements​
+
+Harness recommends `cache.m7g.8xlarge` or higher nodes to ensure the following minimum requirements:
+
+- 8 cores per machine
```
Comment (akshay-khole) [inline tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:16]: @kapilgarg1996  do we have justification for these resources ?

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+## Software requirements​
+
+External database setup requires the following software:
+
+- Supported Redis Version: 6.2
+
+## Supported Redis Configuration
+
+- Master nodes: 1
```
Comment (akshay-khole) [inline tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:32]: Elastic cache terminology is primary and replica

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+2. Create a security group to allow a connection from your Kubernetes cluster to an ElastiCache instance.
+
+## Configure your Harness environment and Helm chart​
+
+After you've set up ElastiCache, you're ready to configure your Harness environment.
+
+To configure your Harness environment and Helm chart, do the following:
+
+1. If you enabled authentication during setup, create a Kubernetes secret in the namespace where Harness Self-Managed Enterprise Edition is installed with the following data.
```
Comment (akshay-khole) [inline tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:66]: we should add explicit commands in document to create secret etc

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+   ```yaml
+   data:
+     root-password: <AUTH_TOKEN>
+     root-username: ""
+   ```
+
+2. Update the following entries in your `override.yaml` file and then upgrade Harness.
+
+   ```yaml
```
Comment (akshay-khole) [inline tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:76]: @brian-f-harness in your assessment is this instruction prescriptive enough ?

Comment (brian-f-harness) [reply tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:76]: @akshay-khole I think it's clear enough. Do you think it'd help if we linked to the Helm upgrade instructions?

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md (hunk 2 above)
Comment (kapilgarg1996) [reply tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:16]: Yes, these come from our prod resources combined for all use cases.

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md (hunk 3 above)
Comment (brian-f-harness) [reply tutorials/self-managed-enterprise-edition/use-aws-elasticache.md]: Thanks @akshay-khole. Updated.

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md (hunk 4 above)
Comment (brian-f-harness) [reply tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:66]: @kapilgarg1996 Can you help with the command here?

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+To configure your Harness environment and Helm chart, do the following:
+
+1. If you enabled authentication during setup, create a Kubernetes secret in the namespace where Harness Self-Managed Enterprise Edition is installed with the following data.
+
+   ```yaml
+   data:
+     root-password: <AUTH_TOKEN>
+     root-username: ""
+   ```
```
Comment (kapilgarg1996) [inline tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:72]:
```suggestion
1. If you enabled authentication during setup, create a Kubernetes secret in the namespace where Harness Self-Managed Enterprise Edition is installed with the following command.

   ```sh
   kubectl create secret generic redis-user-pass \
    --from-literal=root-username="" \
    --from-literal=roo-password='<AUTH_TOKEN>'
   ```
```

File: tutorials/self-managed-enterprise-edition/use-aws-elasticache.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+   ```yaml
+   global:
+     database:
+       redis:
+         installed: false
+         hosts:
+         - <ELASTICACHE_PRIMARY_ENDPOINT_AND_PORT>
+         secretName: "REDIS_SECRET_NAME_CREATED_IN_FIRST_STEP" #leave empty if auth is disabled
+         userKey: "root-username"
+         passwordKey: "root-password"
```
Comment (kapilgarg1996) [inline tutorials/self-managed-enterprise-edition/use-aws-elasticache.md:85]:
```suggestion
         # keep the following 3 values empty if auth is disabled
         secretName: "redis-user-pass" 
         userKey: "root-username"
         passwordKey: "root-password"
```

**Comment by brian-f-harness** [comment]
@akshay-khole Kapil added the commands. Is this now ready to merge?
cc @pratmit @kapilgarg1996

**Review by akshay-khole — APPROVED** [review]

**Pull request merged**